### PR TITLE
Prow: Post a Skipped status to PRs that do not match run_if_changed.

### PR DESCRIPTION
**Pull request by crunk1**

### CODE REVIEW

File: prow/plugins/trigger/pr.go [hunk 1]
```diff
@@ -163,6 +163,13 @@ func buildAll(c client, pr github.PullRequest) error {
 				}
 			}
 			if !job.RunsAgainstChanges(changes) {
+				if err := c.GitHubClient.CreateStatus(org, repo, pr.Head.SHA, github.Status{
```
Comment (rmmh) [inline prow/plugins/trigger/pr.go]: maybe set `skipped = true` here, and change the check below to `if skipped || !job.RunsAgainstBranch...`

**Comment by rmmh** [comment]
Please run ./bump.sh hook to bump the versions.

File: prow/plugins/trigger/pr.go (hunk 1 above)
Comment (crunk1) [reply prow/plugins/trigger/pr.go]: I was thinking about refactoring out the copied code, but couldn't see a good solution. That looks like a good solution.

**Comment by rmmh** [comment]
/lgtm

feel free to squash, or I'll do it when I hit the merge button

**Comment by crunk1** [comment]
I figured you would squash on merge, but I went ahead and squashed.

**Comment by kargakis** [comment]
How come the `needs-ok-to-test` label is still on but tests run?

**Pull request merged**